File: School-Management-System/WebApi/ClientApp/src/app/home/components/fees/model/feeReportRow.viewModel.ts
export interface FeeReportRowViewModel {
  studentEnrollmentId: string;
  studentName: string;
  rollNumber?: number | null;
  totalFees: number;
  totalDiscount: number;
  totalFine: number;
  netFees: number;
  totalPaid: number;
  totalPending: number;
  previousYearPending: number;
  grandTotalPending: number;
  hasPendingFees: boolean;
}

File: School-Management-System/Application/Fees/Dtos/FeeReportRowViewModel.cs
using System;

namespace Application.Fees.Dtos
{
    public class FeeReportRowViewModel
    {
        public Guid StudentEnrollmentId { get; set; }
        public string StudentName { get; set; } = string.Empty;
        public int? RollNumber { get; set; }
        public decimal TotalFees { get; set; }
        public decimal TotalDiscount { get; set; }
        public decimal TotalFine { get; set; }
        public decimal NetFees { get; set; }
        public decimal TotalPaid { get; set; }
        public decimal TotalPending { get; set; }
        public decimal PreviousYearPending { get; set; }
        public decimal GrandTotalPending { get; set; }
        public bool HasPendingFees => TotalPending > 0;
    }
}
